### PR TITLE
Deploy documentation to RTD

### PR DESCRIPTION
**Pull request by aerorahul**
**Description**
This PR adds an Github action that will build and deploy documentation as artifacts for with every push to `develop`.
The artifacts can be downloaded from Actions -> Documentation -> Artifacts

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
To be tested with the github action.
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [X] Any dependent changes have been merged and published

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    branches:
+      - develop
+      - feature/*
+      - main/*
+      - bugfix/*
+      - release/*
+    paths:
+      - docs/**
+  pull_request:
+    types: [opened, reopened, synchronize]
```
Comment (WalterKolczynski-NOAA) [inline .github/workflows/docs.yaml:13]: Not sure if all these should trigger the action. Shouldn't it only be on pushes to develop?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (arunchawla-NOAA) [reply .github/workflows/docs.yaml:13]: If I understand this correctly this should create a documentation build for all the branches that have sphinx documentation and get updated. Where are the documentation files uploaded though?

**Review by arunchawla-NOAA — APPROVED** [review]
Is the build still deployed through RTD?

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply .github/workflows/docs.yaml:13]: L38?

Comment (aerorahul) [reply .github/workflows/docs.yaml:13]: the uploads are as artifacts on github actions.  That is not RTD.  The artifacts will allow you to download the whole documentation.

To answer @WalterKolczynski-NOAA, I believe we want to build the latest documentation (`develop`), but we also want to build documentation for releases for sure.  Not sure about bugfixes and feature branches.

I know RTD does allow one to toggle between documentations between releases and latest.

Comment (WalterKolczynski-NOAA) [reply .github/workflows/docs.yaml:13]: Okay, if RTD can handle branches, I'm satisfied.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by aerorahul** [comment]
> Is the build still deployed through RTD?

Yes.
Nothing at the RTD end has changed.
This PR simply automates the creation of sphinx documentation in the Github action and upload it as artifacts for a-posteriori inspection.

**Pull request merged**